### PR TITLE
Responding to feedback

### PR DESCRIPTION
**Pull request by jmarolf**
fixes for feedback from: https://github.com/dotnet/roslyn/pull/23125

### CODE REVIEW

File: build/Targets/Versions.props [hunk 1]
```diff
@@ -10,10 +10,10 @@
   <PropertyGroup>
     <!-- This is the assembly version of Roslyn from the .NET assembly perspective. It should only be revved during significant point releases. -->
     <!-- PROTOTYPE(NullableReferenceTypes): Revert before merging to master. -->
-    <RoslynAssemblyVersionBase Condition="'$(RoslynAssemblyVersion)' == ''">2.6.1</RoslynAssemblyVersionBase>
+    <RoslynAssemblyVersionBase Condition="'$(RoslynAssemblyVersion)' == ''">42.42.42</RoslynAssemblyVersionBase>
```
Comment (cston) [inline build/Targets/Versions.props]: Why 42.42.42 rather than 2.6.2 or 2.7.0?

Comment (jmarolf) [reply build/Targets/Versions.props]: @jasonmalinowski pointed out that 2.6.1 is going to be a real build number in the near future.  If we are fine with this breaking in 15.5.1 RTM then we can leave it as-is.

**Comment by jmarolf** [comment]
@dotnet/roslyn-infrastructure Looks like the toolset compiler version and the IOperation API being used by our analyzers is out of sync.  Is someone already working on updating our toolset compiler?

**Comment by jmarolf** [comment]
I guess I should ask: Has this already been fixed in master and its just that features/NullableReferenceTypes needs to merge in the necessary upstream changes?

File: build/Targets/Versions.props (hunk 1 above)
Comment (jaredpar) [reply build/Targets/Versions.props]: I feel strongly we should not be using 42.42.42. This is specifically meant for developer builds. Putting it out in the wild means that we're going to be getting Watson's that look a heck of a lot like internal crashes. We should be using 2.6.1 unless there is a specific reason not to. In that case we should not use 42.42.42 as it conflicts with the goal of 42.42.42 which is "build that should never release"

Comment (tmat) [reply build/Targets/Versions.props]: How about 2.7 or 3.0?

Comment (AdamSpeight2008) [reply build/Targets/Versions.props]: How about `0.0.0.0`?

Comment (cston) [reply build/Targets/Versions.props]: Any objections to 2.7?

**Comment by jmarolf** [comment]
ping @jaredpar @jasonmalinowski

**Pull request merged**